clearwater-sip-stress.root/usr/share/clearwater/bin/run_stress

```diff
@@ -0,0 +1,151 @@
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+from random import shuffle
+from subprocess import Popen
+import argparse
+
+# Constants - directories and load profile
+
+SIPP_BIN = "/usr/share/clearwater/bin/sipp"
+SCRIPTS_DIR = "/usr/share/clearwater/sip-stress/"
+
+REGISTER_CSV = "/tmp/registers.csv"
+INVITE_CSV = "/tmp/calls.csv"
+
+BHCA = 1.3
+REREG_PER_HOUR = 2
+
+# Get command-line arguments
+
+parser = argparse.ArgumentParser(description='Run stress.')
+parser.add_argument('domain', help="Deployment's home domain")
+parser.add_argument('subscribers', help='Number of subscribers to emulate', type=int)
+parser.add_argument('duration', help='Number of minutes to run stress for', type=int)
+parser.add_argument('--multiplier', help='Multiplier for the VoLTE load profile (e.g. passing 2 here will mean 2.6 calls and 4 re-registers per sub per hour)', type=int, default=1)
+parser.add_argument('--initial-reg-rate', help='Rate to send in initial, pre-test-run registrations', type=int, default=80)
+parser.add_argument('--target', help='Domain/IP and port to target stress at. Default is sprout.{domain}:5054')
+
+args = parser.parse_args()
+
+if args.target is None:
+    args.target = "sprout.{}:5054".format(args.domain)
+
+# Calculate calls/registers per second based on the input
+CPS = BHCA * args.subscribers * args.multiplier / 3600
+RPS = REREG_PER_HOUR * args.subscribers * args.multiplier / 3600
+
+# Given the duration, how many registers/calls do we need to make?
+REG_MAX = int(args.duration * 60 * RPS)
+CALL_MAX = int(args.duration * 60 * CPS)
+
+SUB_START = 2010000000
+SUB_END = SUB_START + args.subscribers
+
+registers = range(SUB_START, SUB_END)
+callers = range(SUB_START, SUB_END, 2)
+
+# Randomly shuffle the order of registers and calls to reduce the risk of any cache issues
+shuffle(registers)
+shuffle(callers)
+
+# Generate the CSV input files - one for REGISTER messages with one user per
+# line, one for INVITE flows with a caller/callee pair on each line
+with open(REGISTER_CSV, "w") as f:
+    f.write("SEQUENTIAL\n")
+    for s in registers:
+        f.write("{}\n".format(s))
+
+with open(INVITE_CSV, "w") as f:
+    f.write("SEQUENTIAL\n")
+    for s in callers:
+        f.write("{};{}\n".format(s, s+1))
```

Comment:
I was wondering whether there was any value in shuffling the callees too. Currently the caller always calls the same person, right? I can't see immediately why that would be a problem, but it would be so easy to shuffle the callees too that it might make sense to do so.